src/server/object_services/object_server.js

```diff
@@ -1928,11 +1928,13 @@ async function _complete_object_multiparts(obj, multipart_req) {
         }
         md5.update(Buffer.from(mp.md5_b64, 'base64'));
         const mp_parts = parts_by_mp[mp._id.toHexString()];
```

Comment:
I think that you can use this instead:
```const mp_parts = parts_by_mp[mp._id.toHexString()] || [];```

My reasoning is that in other flows we have the same problem but since the return, there is an empty array we just pass the sorting without errors.

```
const map_res = req.rpc_params.multiparts ?
        await _complete_object_multiparts(obj, req.rpc_params.multiparts) :
        await _complete_object_parts(obj);
```
You can see that this returns an empty array in the parts case:
```
    /**
     * @param {nb.ObjectMD} obj
     * @returns {Promise<nb.PartSchemaDB[]>}
     */
    async find_all_parts_of_object(obj) {
        return this._parts.find({ obj: { $eq: obj._id, $exists: true }, deleted: null });
    }
```
That is the reason we aren't failing in other flows.
So I think that instead of changing the logic we can just align with the other case and pass an empty array when we do not have parts.
Another thing that we need to make sure of is that we are not breaking the context.num_parts.
A person can upload 20 parts and only 10 will have an actual size, in that case, the num_parts will be 10 instead of 20 because we won't account for them in the loops.
It is a bit confusing...

Reply:
@jeniawhite fixed and added a unit-test. please re-review and approve